ndcube/extra_coords/table_coord.py

```diff
@@ -136,7 +135,10 @@ def _generate_tabular(lookup_table, interpolation='linear', points_unit=u.pix, *
         raise TypeError("lookup_table must be a Quantity.")  # pragma: no cover
 
     ndim = lookup_table.ndim
-    TabularND = tabular_model(ndim, name=f"Tabular{ndim}D")
+    if ndim == 1:
+        TabularND = Tabular1D
+    elif ndim == 2:
+        TabularND = Tabular2D
```

Comment:
I think this should have an else for >2D. Also a comment saying that this is for asdf would be good for future reference.

Reply:
Sure